src/main/java/net/minestom/server/network/packet/server/play/DisconnectPacket.java

```diff
@@ -7,16 +7,32 @@
 import org.jetbrains.annotations.NotNull;
 
 public class DisconnectPacket implements ServerPacket {
+    public String message;
 
-    public JsonMessage message; // Only text
+    /**
+     * @deprecated Use {@link #message}
+     */
+    @Deprecated public JsonMessage messageJson;
```

Comment:
If you are replacing a JsonMessage by a String, do delete the JsonMessage field, there is no point in keeping it.
You are already breaking the API by changing the type, go all the way!

Same goes for other JsonMessages inside other packet classes.

Reply:
I was keeping the old fields in order to maintain compatibility - in the writing methods it will write the `messageJson` if it exists. I'm happy to remove all these if needs be but I think it might be better to remove them alongside the removal of other deprecated methods at a later point in order to give people a bit of time to change over.